web-app/src/pages/Dashboard/container.js

```diff
@@ -32,12 +33,18 @@ const DashboardContainer = Wrapped =>
         activeEventTypeIds: [],
         activeHolidayStatusIds: [],
         activeEmployee: -1,
-        eventView: eventsView.PERSONAL_EVENTS,
       };
     }
 
     componentDidMount() {
+      const { setEventView } = this.props;
+      setEventView(eventsView.PERSONAL_EVENTS);
       this.fetchEvents(eventsView.PERSONAL_EVENTS, true);
+
+      // This will be called when the user navigates away from the page.
+      this.unlisten = this.props.history.listen(() => {
```

Comment:
Could we use something like removeListener or removeX or something here. Just the naming of it seems a bit off.

Reply:
Also why can't you stick the function inside componentWillUnmount?

Reply:
@girwin575 With react-router-dom the pages you've previously visited stay mounted.

Reply:
Renamed to `removeRouterListener`

Reply:
Cool!